instrumentation/racecar/lib/opentelemetry/instrumentation/racecar/process_message_subscriber.rb

```diff
@@ -54,6 +56,15 @@ def finish(name, id, payload)
         OpenTelemetry::Context.detach(token)
         OpenTelemetry::Context.detach(parent_token)
       end
+
+      def extract_message_key(key)
+        # skip encode if already valid utf8
+        return key if key.nil? || (key.encoding == Encoding::UTF_8 && key.valid_encoding?)
+
+        key.encode(Encoding::UTF_8)
+      rescue Encoding::UndefinedConversionError
+        nil
```

Comment:
Do you think users would be interested in cases where binary keys where being used and report this error using `OpenTelemetry.handle_error(exception: e)`?

Reply:
I'm not sure tbh. The trigger for raising this PR is that we're raising errors already in our OTLP exporters and we'd like that to stop. If we were to raise an error here we'd just be exchanging a vague error for a not so vague one. 

The messages we consume are working as intended and we have no intention to migrate these keys to a different format (I'm fairly certain it would break some upstream services) so this would just cause us continuous error spam.

I believe the correct move is to update the Otel specification conventions for Kafka message to provide guidance on how to convert a binary value into a recognisable key. I'm not sure how to navigate this process though.